### DISCUSSION TITLE
lab SDRAM 問題

### DISCUSSION THREAD

**Discussion by DonghanTsai** [bol-edu/HLS-SOC-Discussions#182] [General]
測試了一下發現dram是可以連續read data的
感覺這個特性跟真的dram有點不同
請問我們可以利用這樣的特性嗎
還是必須等到dram讀出一個data之後才能去read下一個

**Comment by boledulab** [comment]
能 PO 出 waveform, 如此才能判斷是否是正確的行為。DRAM 是有 burst read/write 的能力。

**Comment by DonghanTsai** [comment]
應該沒有使用到burst mode，像是如果我故意在READ state多停留一個cycle (傳給dram的訊號，地址等也維持不變)，我可以觀察到dram將會在3個cycle之後連續吐出兩個一樣的值給我，那我們能夠利用這樣的特性，在dram上一筆資料還沒read出來的情況下再跟dram要下一個資料嗎?

**Comment by alankuo04** [comment]
同學你好

Lab SDRAM 裡面提到的 prefetch 就是要在你說的情況下讀取下一個位置的資料，也就是在前一筆 read 還沒讀取回來的情況下，再跟 dram 要下一個 read 的資料
因為 wishbone 的 interface 的原因，才會讓原本可以在 dram 讀取出資料前就繼續發送下一筆的 read 指令無法被使用，因此這個 lab 就是希望你們能夠把這個功能加進來，並且比較一下跟原本的版本之間的差異

**Reply by DonghanTsai** [reply]
那我可以理解為這個lab的目標是在硬體上只更改sdram controller的情況下去加速read sdram的過程嗎? 
如果是這樣那僅僅只透過加入prefetch下一個address的方式我們可能就已經可以將一個平均的read時長縮短為2個cycle (第1個收input, 第2個馬上輸出output)，那這樣我們還能再透過更改row, column, bank address的排列方式或者投影片上提到的bank interleave又可以再加速些什麼呢?

**Reply by alankuo04** [reply]
> 那我可以理解為這個lab的目標是在硬體上只更改sdram controller的情況下去加速read sdram的過程嗎?

對，SDRAM加速主要就是這個部分

> 如果是這樣那僅僅只透過加入prefetch下一個address的方式我們可能就已經可以將一個平均的read時長縮短為2個cycle (第1個收input, 第2個馬上輸出output)，那這樣我們還能再透過更改row, column, bank address的排列方式或者投影片上提到的bank interleave又可以再加速些什麼呢?

這部分應該就沒有加速，主要是希望同學可以嘗試修改 linker 來把資料放在不同的 bank 當中，可參考 #159
